pulp/pulp.py

```diff
@@ -230,7 +230,10 @@ def __ge__(self, other):
         return LpAffineExpression(self) >= other
 
     def __eq__(self, other):
-        return LpAffineExpression(self) == other
+        if isinstance(other, LpElement) and self.hash == other.hash:
+            return not self.__ne__(other)
```

Comment:
I'm not sure why you're doing this but this i think this does not do what you want it to do.

`__ne__` returns a boolean value (becuse "!=" constraints do not exist in MIP problems).
`__eq__` returns an `LpAffineExpression.__eq__` function, which itself returns a `LpConstraint`.

So... you cannot return a boolean when you should be returning a `LpConstraint` object.

Reply:
Indeed... I clearly missed that point.
However  __eq__ function will still be called in case you are storing severel LpVariable in a dict that have some hash collision... causing the StackOverflow based on current __eq__ implementation returning a new object...
Perharps a new definition of the hash function would be a better fix?